src/index.ts

```diff
@@ -204,7 +204,7 @@ export class Factory<T> extends Faker {
         }
         if (isRecord(value)) {
             return Object.fromEntries(
-                Object.entries(value).map(([k, v]) => [k, this.parseValue(v)]),
+                Object.entries(value as Record<string, unknown>).map(([k, v]) => [k, this.parseValue(v)]),
```

Comment:
this shouldnt be required - you have a type predicate in isRecord above, which means this should be properly typed here.